pipe/read.py

```diff
@@ -1,311 +1,322 @@
-    c.append(fits.Column(name='YC', format='D', unit='pix', array=yc))
-    c.append(fits.Column(name='FLAG', format='I', array=flag))
-    for n in range(w.shape[1]):
-        c.append(fits.Column(name='U{:d}'.format(n), format='D', array=w[:,n]))
-    c.append(fits.Column(name='thermFront_2', format='D', array=thermFront_2))
-    tab = fits.BinTableHDU.from_columns(c, header=header)
-    tab.writeto(filename, overwrite=True, checksum=True)
-
-
-def save_binary_eigen_fits(filename, t, bjd, sc0, sc1, bg, roll, 
-                           xc0, yc0, xc1, yc1, flag, chi2,
-                           w0, w1, thermFront_2, header):
-    """Save lightcurve data from both componentes of a binary, as defined
-    by arguments to fits table in binary format. Coefficients for the
-    principle components of both stars from the PSF eigen analysis are
-    also added, as well as the thermFront_2 values, to be used in
-    de-correlations.
-    """
-    c = []
-    c.append(fits.Column(name='MJD_TIME', format='D', unit='day', array=t))
-    c.append(fits.Column(name='BJD_TIME', format='D', unit='day', array=bjd))
-    c.append(fits.Column(name='FLUX0', format='D', unit='electrons', array=sc0))
-    c.append(fits.Column(name='FLUX1', format='D', unit='electrons', array=sc1))
-    c.append(fits.Column(name='BG', format='D', unit='electrons/pix', array=bg))
-    c.append(fits.Column(name='CHI2', format='D', array=chi2))
-    c.append(fits.Column(name='ROLL', format='D', unit='deg', array=roll))
-    c.append(fits.Column(name='XC0', format='D', unit='pix', array=xc0))
-    c.append(fits.Column(name='YC0', format='D', unit='pix', array=yc0))
-    c.append(fits.Column(name='XC1', format='D', unit='pix', array=xc1))
-    c.append(fits.Column(name='YC1', format='D', unit='pix', array=yc1))
-    c.append(fits.Column(name='FLAG', format='I', array=flag))
-    for n in range(w0.shape[1]):
-        c.append(fits.Column(name='U{:d}'.format(n), format='D', array=w0[:,n]))
-    for n in range(w1.shape[1]):
-        c.append(fits.Column(name='W{:d}'.format(n), format='D', array=w1[:,n]))
-    c.append(fits.Column(name='thermFront_2', format='D', array=thermFront_2))
-    tab = fits.BinTableHDU.from_columns(c, header=header)
-    tab.writeto(filename, overwrite=True, checksum=True)
-
-
-def save_txt(filename, t, flux, err, bg, roll, xc, yc):
-    """Save lightcurve to textfile according to arrays
-    defined by arguments
-    """
-    X = np.array([t, flux, err, bg, roll, xc, yc]).T
-    fmt = '%26.18e'
-    np.savetxt(filename, X=X, fmt=fmt)
-
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Mar  4 16:00:21 2020
+
+@author: Alexis Brandeker, alexis@astro.su.se
+
+A collection of routines that read data from CHEOPS fits format
+files, and typically returns numpy arrays.
+
+Also contains routines that save data in fits or text formats.
+
+"""
+
+import os
+import numpy as np
+from scipy import interpolate
+from astropy.io import fits
+
+fits.Conf.use_memmap = False
```

Comment:
Without this line, the astropy machinery will use [memmap](https://docs.astropy.org/en/stable/io/fits/#working-with-large-files), which does not close the filestreams and causes the tests to fail on Windows.